eclair-core/src/test/scala/fr/acinq/eclair/payment/PostRestartHtlcCleanerSpec.scala

```diff
@@ -419,6 +421,45 @@ class PostRestartHtlcCleanerSpec extends TestKitBaseClass with FixtureAnyFunSuit
     channel_upstream_3.expectNoMsg(100 millis)
   }
 
+  test("handle a channel relay htlc-fail") { f =>
+    import f._
+
+    val testCase = setupChannelRelayedPayments(nodeParams)
+    val relayer = f.createRelayer()
+    register.expectNoMsg(100 millis)
+
+    // we need a reference to the post-htlc-restart child actor
+    sender.send(relayer, Relayer.GetChildActors(sender.ref))
+    val postRestartHtlcCleaner = sender.expectMsgType[Relayer.ChildActors].postRestartCleaner
+
+    sender.send(relayer, buildForwardFail(testCase.downstream, testCase.origin))
+    register.expectMsgType[Register.Forward[CMD_FAIL_HTLC]]
+
+    register.expectNoMsg(100 millis) // the payment has already been fulfilled upstream
```

Comment:
nit: comment doesn't seem to be accurate? But maybe it would be interesting to re-send `buildForwardFail(testCase.downstream, testCase.origin)` a second time to the `relayer` and assert it's a noop?

Reply:
Done with 6be5ba134.